modules/stack/main.tf

```
@@ -21,6 +21,24 @@ resource "spacelift_stack" "default" {
   terraform_workspace = var.terraform_workspace
 }
 
+resource "spacelift_run" "this" {
+  count = var.enabled && var.commit_sha != null ? 1 : 0
```

If we do not want the `spacelift_run` resource based on the `var.commit_sha`, then we can make it based on a new boolean called `var.spacelift_run`.

```suggestion
  count = var.enabled && var.spacelift_run ? 1 : 0
```